robotoff/workers/tasks/import_image.py

```diff
@@ -47,37 +68,95 @@ def run_import_image_job(
         return
 
     with db:
-        with db.atomic():
-            save_image(barcode, source_image, product, server_domain)
-            import_insights_from_image(
-                barcode, image, source_image, ocr_url, server_domain
-            )
-        with db.atomic():
-            # Launch object detection in a new SQL transaction
-            run_object_detection(barcode, image, source_image, server_domain)
+        save_image(barcode, source_image, product, server_domain)
+
+    enqueue_job(
+        import_insights_from_image,
+        high_queue,
+        barcode=barcode,
+        image_url=image_url,
+        ocr_url=ocr_url,
+        server_domain=server_domain,
+    )
+    enqueue_job(
+        run_logo_object_detection,
+        high_queue,
+        barcode=barcode,
+        image_url=image_url,
+        server_domain=server_domain,
+    )
+    enqueue_job(
+        run_nutrition_table_object_detection,
+        high_queue,
+        barcode=barcode,
+        image_url=image_url,
+        server_domain=server_domain,
+    )
 
 
 def import_insights_from_image(
     barcode: str,
-    image: Image.Image,
-    source_image: str,
+    image_url: str,
     ocr_url: str,
     server_domain: str,
 ):
-    predictions_all = get_predictions_from_image(barcode, image, source_image, ocr_url)
+    image = get_image_from_url(image_url, error_raise=False, session=http_session)
+
+    if image is None:
+        logger.info(f"Error while downloading image {image_url}")
+        return
+
+    source_image = get_source_from_url(image_url)
+    predictions = extract_ocr_predictions(
+        barcode, ocr_url, DEFAULT_OCR_PREDICTION_TYPES
+    )
+    if any(
+        prediction.value_tag == "en:nutriscore"
+        and prediction.type == PredictionType.label
+        for prediction in predictions
+    ):
+        enqueue_job(
+            run_nutriscore_object_detection,
+            high_queue,
+            barcode=barcode,
+            image_url=image_url,
+            server_domain=server_domain,
+        )
```

Comment:
I would set a boolean variable to True and launch the job at the end.

Doing that you augment the risk of conflict (even with a lock, it's better to avoid it).